### PR TITLE
Update spaceflights tutorial and various other minor improvements

### PR DESCRIPTION
**Pull request by AntonyMilneQB**
### Spaceflights tutorial
Tidied up code throughout. Mostly minor stylistic changes, e.g. to show best practice use of `pandas` and consistent naming of nodes. The most significant changes are:
* features used in the model expanded (previously these only came from one table in the raw data, which rendered the master table part of the pipeline kind of pointless)
* features moved to parameters file
* pandas operations vectorised rather than using `apply`

Tested all works well by doing `kedro new --starter=spaceflights` and running `kedro run`, `kedro ipython` and `kedro build-docs`.

### Other changes
Again, many minor stylistic changes for improved consistency. Most significant changs are:
* added a README file to the repo
* added `configure_project(metadata.package_name)` to `reload_kedro` in each starter

### Still to do
Update documentation with new code examples.

### CODE REVIEW

**Comment by lorenabalan** [comment]
When you have a minute, can you also address https://github.com/quantumblacklabs/kedro-starters/issues/21 please? :)

File: README.md [hunk 1]
```diff
@@ -1,3 +1,14 @@
...
+```
+
+The following aliases are available:
+* [Alias `mini-kedro`](mini-kedro): A minimum setup to use the traditional [Iris dataset](https://www.kaggle.com/uciml/iris) with Kedro's [`DataCatalog`](https://kedro.readthedocs.io/en/stable/05_data/01_data_catalog.html), which is a core component of Kedro. This starter is of use in the exploratory phase of a project. For more information, please read the [Mini-Kedro](https://kedro.readthedocs.io/en/stable/04_kedro_project_setup/04_mini_kedro.html) guide.
+* [Alias `pandas-iris`](pandas-iris): The [Kedro Iris dataset example project](https://kedro.readthedocs.io/en/stable/02_get_started/05_example_project.html)
+* [Alias `pyspark-iris`](pyspark-iris): An alternative Kedro Iris dataset example, using [PySpark](https://kedro.readthedocs.io/en/stable/11_tools_integration/01_pyspark.html)
+* [Alias `pyspark`](pyspark): The configuration and initialisation code for a [Kedro pipeline using PySpark](https://kedro.readthedocs.io/en/stable/11_tools_integration/01_pyspark.html)
+* [Alias `spaceflights`](spaceflights): The [spaceflights tutorial](https://kedro.readthedocs.io/en/stable/03_tutorial/01_spaceflights_tutorial.html) example code
```
Comment (MerelTheisenQB) [inline README.md:14]: Very nice!👏

File: spaceflights/README.md [hunk 2]
```diff
@@ -1,16 +1,16 @@
-# Kedro starter for the Spaceflights tutorial (`kedro-starter-spaceflights`)
+# The `spaceflights` Kedro starter
 
 ## Overview
 
-This is a tutorial project, which was generated using `Kedro 0.16.5`. It is the completed version of the [Space Flights tutorial](https://kedro.readthedocs.io/en/stable/03_tutorial/01_spaceflights_tutorial.html) described in the [online Kedro documentation](https://kedro.readthedocs.io) and includes the data necessary to run the project.
+This is a tutorial project, which was generated using `Kedro 0.16.5`. It is the completed version of the [spaceflights tutorial](https://kedro.readthedocs.io/en/stable/03_tutorial/01_spaceflights_tutorial.html) described in the [online Kedro documentation](https://kedro.readthedocs.io) and includes the data necessary to run the project.
```
Comment (MerelTheisenQB) [inline spaceflights/README.md:5]: Is this project really generated with Kedro `0.16.5`? The starters should just pick up whatever version of Kedro you're using, so this sentence is quite confusing. We should probably just delete it.

File: spaceflights/README.md [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 
 The tutorial works through the steps necessary to create this project. To learn the most about Kedro, we recommend that you start with a blank template as the tutorial describes, and follow the workflow. However, if you prefer to read swiftly through the documentation and get to work on the code, you may want to generate a new Kedro project using this [starter](https://kedro.readthedocs.io/en/stable/02_get_started/06_starters.html) because the steps have been done for you.
 
 To use this starter, create a new Kedro project using the commands below. To make sure you have the required dependencies, run it in your virtual environment (see [our documentation about virtual environments](https://kedro.readthedocs.io/en/stable/02_get_started/01_prerequisites.html#virtual-environments) for guidance on how to get set up):
 
 ```bash
 pip install kedro==0.16.5
```
Comment (MerelTheisenQB) [inline spaceflights/README.md:12]: I would also remove the pinning of kedro to `0.16.5` here.

**Review by MerelTheisenQB — APPROVED** [review]
Great work! 😄

File: spaceflights/README.md (hunk 2 above)
Comment (AntonyMilneQB) [reply spaceflights/README.md]: Very good point. Done in a6908c2

**Review by limdauto — APPROVED** [review]
Thank you

File: pyspark-iris/{{ cookiecutter.repo_name }}/.ipython/profile_default/startup/00-kedro-init.py [hunk 4]
```diff
@@ -52,6 +53,7 @@ def reload_kedro(path, line=None, env: str = None, extra_params: Dict[str, Any]
         for module in to_remove:
             del sys.modules[module]
 
+        configure_project(metadata.package_name)
```
Comment (lorenabalan) [inline pyspark-iris/{{ cookiecutter.repo_name }}/.ipython/profile_default/startup/00-kedro-init.py:56]: Nice thank you, these definitely need to go in the `0.17.1` release of the starters.

File: spaceflights/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/pipelines/data_engineering/README.md [hunk 5]
```diff
@@ -37,18 +34,18 @@ This modular pipeline preprocesses the raw data (`preprocessing_companies` and `
...
-| Description | Cleansed and normalised version of the `shuttles` dataset |
+| Description | Preprocessed version of the `shuttles` dataset |
 
 ### `master_table`
 
 |      |                    |
 | ---- | ------------------ |
 | Type | `pandas.DataFrame` |
-| Description | A combined dataset from preprocessed companies and shuttle datasets, and source reviews data |
+| Description | A combined dataset containing data on shuttles, with company and reviews information joined in |
```
Comment (lorenabalan) [inline spaceflights/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/pipelines/data_engineering/README.md:51]: Minor thing, just cause sth triggers me about sometimes about ending a sentence in a preposition. 😂  Feel free to ignore if not bothered.

```suggestion
| Description | A combined dataset containing data on shuttles, joined with company and reviews information |
```

File: spaceflights/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/pipelines/data_science/README.md [hunk 6]
```diff
@@ -17,14 +16,14 @@ This modular pipeline:
 |      |                    |
 | ---- | ------------------ |
 | Type | `pandas.DataFrame` |
-| Description | A combined dataset from preprocessed companies and shuttle datasets, and source reviews data |
+| Description | A combined dataset containing data on shuttles, with company and reviews information joined in |
```
Comment (lorenabalan) [inline spaceflights/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/pipelines/data_science/README.md:19]:
```suggestion
| Description | A combined dataset containing data on shuttles, joined with company and reviews information |
```

**Review by lorenabalan — APPROVED** [review]
Stellar work! ✨ ⭐

**Pull request merged**